src/main/java/io/github/thebusybiscuit/slimefun4/implementation/items/electric/machines/AutoBrewer.java

```diff
@@ -0,0 +1,201 @@
+
+        if (input1.getType().name().endsWith("POTION") || input2.getType().name().endsWith("POTION")) {
+            boolean slot = input1.getType().name().endsWith("POTION");
+            ItemStack pItem = slot ? input1 : input2;
+            ItemStack iItem = slot ? input2 : input1;
+
+            PotionMeta potion = (PotionMeta) pItem.getItemMeta();
+            if (potion == null) return null;
+            PotionData potionData = potion.getBasePotionData();
+
+            ItemStack output;
+            if (potionData.getType() == PotionType.WATER) {
+                if (iItem.getType() == Material.FERMENTED_SPIDER_EYE) {
+                    potion.setBasePotionData(new PotionData(PotionType.WEAKNESS, false, false));
+                    output = new ItemStack(pItem.getType());
+
+                } else if (iItem.getType() == Material.NETHER_WART) {
+                    potion.setBasePotionData(new PotionData(PotionType.AWKWARD, false, false));
+                    output = new ItemStack(pItem.getType());
+
+                } else if (pItem.getType() == Material.POTION && iItem.getType() == Material.GUNPOWDER) {
+                    output = new ItemStack(Material.SPLASH_POTION);
+
+                } else if (pItem.getType() == Material.SPLASH_POTION && iItem.getType() == Material.DRAGON_BREATH) {
+                    output = new ItemStack(Material.LINGERING_POTION);
+
+                } else return null;
+
+            } else {
+                if (iItem.getType() == Material.FERMENTED_SPIDER_EYE) {
+                    potion.setBasePotionData(new PotionData(fermSpiderEye.get(potionData.getType()), false, false));
+                    output = new ItemStack(pItem.getType());
+
+                } else if (iItem.getType() == Material.REDSTONE) {
+                    potion.setBasePotionData(new PotionData(potionData.getType(), true, potionData.isUpgraded()));
+                    output = new ItemStack(pItem.getType());
+
+                } else if (iItem.getType() == Material.GLOWSTONE_DUST) {
+                    potion.setBasePotionData(new PotionData(potionData.getType(), potionData.isExtended(), true));
+                    output = new ItemStack(pItem.getType());
+
+                } else if (potionData.getType() == PotionType.AWKWARD && potionRecipes.containsKey(iItem.getType())) {
+                    potion.setBasePotionData(new PotionData(potionRecipes.get(iItem.getType()), false, false));
+                    output = new ItemStack(pItem.getType());
+
+                } else return null;
+            }
+
+            output.setItemMeta(potion);
+
+            return new MachineRecipe(30, new ItemStack[]{input1, input2}, new ItemStack[]{output});
+        } else return null;
+    }
+
+    @Override
+    public String getInventoryTitle() {
+        return "&6Auto-Brewer";
+    }
+
+    @Override
+    public ItemStack getProgressBar() {
+        return new ItemStack(Material.CARROT_ON_A_STICK);
```

Comment:
Why a carrot on a stick?

Reply:
because I thought it looks similar to brewing stand. any better idea?

Reply:
Nah, not really :/

Reply:
if you don't have a better idea why not keep it?